File: src/ShortStackLib/Cmdlets/NewStack.cs
using System;
using System.Management.Automation;

namespace Microsoft.Tools.Productivity.ShortStack
{
    /// <summary>
    /// <para type="synopsis">Start a new set of stacked pull requests.</para>
    /// <para type="description">This is the first command to run.</para>
    /// </summary>    
    [Cmdlet(VerbsCommon.New, "Stack")]
    [OutputType(typeof(StackInfo))]
    public class NewStack : ShortStackCmdletBase
    {
        /// <summary>
        /// <para type="description">The name for this new stack.  Choose a short, descriptive name
        /// with letters, numbers, and underscores.</para>
        /// </summary>
        [Parameter(Position = 1, Mandatory = false)]
        public string Name { get; set; }

        /// <summary>
        /// <para type="description">The origin for this stack.  The default is 'master'.</para>
        /// </summary>
        [Parameter(Position = 2)]
        public string DesiredOrigin { get; set; } = "master"; // This is the default value

        /// <summary>
        /// ProcessRecord
        /// </summary>
        protected override void ProcessRecord()
        {
            ProcessSafely(() =>
            {
                Handler.NewStack(Name, DesiredOrigin);

                Print(ConsoleColor.Green, "===================================");
                Print(ConsoleColor.Green, "---  Your new stack  is ready!  ---");
                Print(ConsoleColor.Green, "===================================");
                Print("Next steps:");
                Print("    1) Commit changes. Keep your changes simple becaue making new stack levels is easy");
                Print("    2) Push your change:  Push-Changes");
                Print("That's it!  Shortstack will auto-create a pull request and a new stack level.");

                if (ObjectOutput.IsPresent)
                {
                    WriteObject(GetCurrentStackLevel());
                }

            });
        }
    }
}

File: nodejs/src/actions/newStack.ts
import { ShortStackOptions } from "../ShortStackOptions";
import { ShortStackProcessor } from "../models/ShortStackProcessor";
import chalk from "chalk";
import { GitHelper } from "../helpers/gitHelper";

//------------------------------------------------------------------------------
// Start a new stack or extend the current stack 1 level
//------------------------------------------------------------------------------
export function newStack(options: ShortStackOptions)
{
    const handler = new ShortStackProcessor(new GitHelper());
    handler.createNewStack(options.stackName, options.stackOrigin);

    console.log(chalk.greenBright("==================================="));
    console.log(chalk.greenBright("---  Your new stack  is ready!  ---"));
    console.log(chalk.greenBright("==================================="));
    console.log("Next steps:");
    console.log("    1) Commit changes. Keep your changes simple becaue making new stack levels is easy");
    console.log("    2) Push your change:  Push-Changes");
    console.log("That's it!  Shortstack will auto-create a pull request and a new stack level.");

    // if (ObjectOutput.IsPresent)
    // {
    //     WriteObject(GetCurrentStackLevel());
    // }

    

}